proposals/lip-0051.md

```diff
@@ -1143,26 +1090,31 @@ transferCrossChain(timestamp,
     tokenID = canonicalTokenID(tokenID)
     chainID = tokenID.chainID
     localID = tokenID.localID
+
+    # basic formating checks
     if (amount < 0
-        or chainID not in [CHAIN_ID_ALIAS_NATIVE, MAINCHAIN_ID, receivingChainID]
-        or (escrowTerminated(receivingChainID) == True
-            and (chainID == CHAIN_ID_ALIAS_NATIVE or chainID == MAINCHAIN_ID))
         or length(data) > MAX_DATA_LENGTH
         or length(senderAddress) != ADDRESS_LENGTH
         or length(recipientAddress) != ADDRESS_LENGTH
         or availableBalance(senderAddress, tokenID) < amount):
         transferCrossChain fails
 
+    # transfer is only possible for tokens native to either sending or receiving chain
+    # and if there is a direct channel, in which case a forward message will be sent
+    if (chainID not in [CHAIN_ID_ALIAS_NATIVE, receivingChainID]
+        and interoperability.isDirectChannelActive(chainID) == False):
```

Comment:
Shouldn't it be an `or` between the two conditions?

Reply:
I think here it is correct. You can always send tokens native to the sending or receiving chain. Or, you can send tokens if you have a direct channel (this is how you can send LSK everywhere).

Reply:
Ah, sorry, my initial comment was due to misreading/misunderstanding the function `isDirectChannelActive`.

However, now if  `interoperability.isDirectChannelActive(chainID) == True` I can actually transfer any token? In particular, before we checked that only `LSK` could be transferred to the mainchain.

Reply:
Yes, you can transfer `LSK` through the mainchain, and in the same way, if you have a direct channel with chainA, you can transfer any tokens native to chainA through chainA.

Reply:
As discussed, we now consider allow token transfers from any chain to any chain, using the forward message to the native chain if necessary.

Reply:
After re-dicussion, we concluded to only allow forwarding of `LSK` tokens as there can be issues with the type of token used for the message fee in other cases.